cmd/docker-mcp/secret-management/provider/desktop.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/docker/mcp-gateway/pkg/desktop"
+)
+
+// DesktopProvider implements SecretProvider using Docker Desktop's secrets API.
+type DesktopProvider struct {
+	client *desktop.Secrets
+}
+
+// NewDesktopProvider creates a new DesktopProvider.
+func NewDesktopProvider() *DesktopProvider {
+	return &DesktopProvider{
+		client: desktop.NewSecretsClient(),
+	}
+}
+
+// GetSecret retrieves a secret from Docker Desktop.
+func (d *DesktopProvider) GetSecret(ctx context.Context, name string) (string, error) {
+	// Docker Desktop doesn't have a direct GetSecret method, so we list all secrets
```

Comment:
The reason for this was that it's going to be possible for the gateway to mount the secret in the container without ever actually knowing the value of the secret (through Docker engine support for secure secret mounts). We need to talk about how to make this available to Docker Community Edition.